jobs/buildhub/to_kinto.py

```diff
@@ -155,16 +102,53 @@ def fetch_existing(
             )
             previous_run_etag = '"%s"' % highest_timestamp
 
-    new_records = client.get_records(
-        _since=previous_run_etag,
-        pages=float('inf')
-    )
-
-    for record in new_records:
-        records[record['id']] = [
-            record['last_modified'],
-            hash_record(record)
-        ]
+    # The reason we can't use client.get_records() is because it is not
+    # an iterator and if the Kinto database has 1M objects we'll end up
+    # with a big fat Python list of 1M objects that has repeatedly caused
+    # OOM errors in our stage and prod admin nodes.
+    if previous_run_etag:
+        # However, we can use it if there was a previous_run_etag which
+        # means we only need to extract a limited amount of records. Not
+        # the whole Kinto database.
+        new_records_batches = [client.get_records(
+            _since=previous_run_etag,
+            pages=float('inf')
+        )]
+    else:
+        def new_records_iterator():
+            params = {'_since': None}
+            next_url = None
+            while True:
+                if next_url is None:
+                    endpoint = client.get_endpoint('records')
```

Comment:
why have endpoint and next_url variables instead of just one initialized out of the loop?

Reply:
I refactored it a little so there's just one variable name that gets reused. 

The `endpoint` (when I print it) looks like this:
```
ENDPOINT /buckets/build-hub/collections/releases/records
ENDPOINT http://localhost:8888/v1/buckets/build-hub/collections/releases/records?_limit=10000&_token=eyJsYXN0X3JlY29yZCI6eyJsYXN0X21vZGlmaWVkIjoxNTI1MzY1MzUyODkyfSwib2Zmc2V0IjoxMDAwMCwibm9uY2UiOiJwYWdpbmF0aW9uLXRva2VuLWZlMjJlNzdmLTg0OWQtNGRjOC1hOTI5LTI4NDNjMWY2MWUzOSJ9
ENDPOINT http://localhost:8888/v1/buckets/build-hub/collections/releases/records?_limit=10000&_token=eyJsYXN0X3JlY29yZCI6eyJsYXN0X21vZGlmaWVkIjoxNTI0NTg5MjY2MjU3fSwib2Zmc2V0IjoyMDAwMCwibm9uY2UiOiJwYWdpbmF0aW9uLXRva2VuLWIyZjA4OTU5LTU4NTgtNGMzNC1iOGUxLWI0Yjc0Zjk2Njg3MSJ9
ENDPOINT http://localhost:8888/v1/buckets/build-hub/collections/releases/records?_limit=10000&_token=eyJsYXN0X3JlY29yZCI6eyJsYXN0X21vZGlmaWVkIjoxNTI0MjQxMzYwNTI5fSwib2Zmc2V0IjozMDAwMCwibm9uY2UiOiJwYWdpbmF0aW9uLXRva2VuLTZkNjQ3NGMwLTkzMTctNDYwNS05ZjU2LTQ1OTg2NWFhYmUxYSJ9
...
```